app/models/relationship.rb

```diff
@@ -0,0 +1,49 @@
+class Relationship < ApplicationRecord
+  belongs_to :relationship_type
+  belongs_to :parent, class_name: :Person
+  belongs_to :child, class_name: :Person
+
+  validate :no_self_relationships
+  validate :permission_check
+
+  attr_accessor :start_node, :node_occupant, :created_by
+
+  def reify
+    raise ArgumentError unless start_node.present? && node_occupant.present?
+    
+    type_id, side = start_node.split(':')
+    self.relationship_type = RelationshipType.find(type_id)
+    
+    if side == 'p'
+      self.parent = node_occupant
+    elsif side == 'c'
+      self.child = node_occupant
+    else
+      raise ArgumentError
+    end
+  end
+
+  private
+
+  def no_self_relationships
+    errors.add(:parent, "cannot be the same person as the child") if parent != nil && parent == child
```

Comment:
I feel like it would look better UI wise and make more sense to have the error show up for both the parent and the child. Currently, it shows up on the parent even if the issue is with the child.